src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java

```diff
@@ -1,64 +1,112 @@
-        intakeMotorLeft = new CANSparkMax(intakeMotorLeftID, MotorType.kBrushless);
-
-        conveyorBeltOne = new CANSparkMax(conveyorBeltOneID, MotorType.kBrushless);
-        conveyorBeltTwo = new CANSparkMax(conveyorBeltTwoID, MotorType.kBrushless);
-
-        intakeMotorBack.follow(intakeMotorFront);
-        intakeMotorLeft.follow(intakeMotorFront);
-        intakeMotorRight.follow(intakeMotorFront);
-
-    }
-    
-    public void intake(double speed, CANSparkMax motor){
-        motor.set(speed);
-    }
-
-    public void stopMotor(CANSparkMax motor){
-        motor.set(0);
-    }
-
-    public void spitOut(){
-        intakeMotorFront.set(INTAKE_FAST_OUT_SPEED);
-    }
-
-    
-    
-    /*
-    public void takeIn(double speed, CANSparkMax motor, CANSparkMax motor2){
-        motor.set(speed);
-        motor2.set(speed); 
+	public static final double INGEST_SPEED = 0.2;
+	public static final double INDEX_SPEEED = 0.2;
+	public static final double FEEDER = 0.25;
+	public static final double DEEPS = 2.0;
+
+	// 'skirt' intake
+	private final CANSparkFlex intakeMotorFront;
+	private final CANSparkFlex intakeMotorBack;
+	private final CANSparkFlex intakeMotorLeft;
+	private final CANSparkFlex intakeMotorRight;
+
+	private final CANSparkFlex injestMotorLeft;
+	private final CANSparkFlex injestMotorRight;
+
+	private final CANSparkFlex indexMotorTop;
+	private final CANSparkFlex indexMotorMiddle;
+	private final CANSparkFlex indexMotorBottom;
+
+	public IntakeSubsystem() {
+
+		intakeMotorFront = new CANSparkFlex(INTAKE_MOTOR_FRONT, MotorType.kBrushless);
+		intakeMotorBack = new CANSparkFlex(INTAKE_MOTOR_BACK, MotorType.kBrushless);
+		intakeMotorLeft = new CANSparkFlex(INTAKE_MOTOR_LEFT, MotorType.kBrushless);
+		intakeMotorRight = new CANSparkFlex(INTAKE_MOTOR_RIGHT, MotorType.kBrushless);
+
+		injestMotorLeft = new CANSparkFlex(INJEST_MOTOR_LEFT, MotorType.kBrushless);
+		injestMotorRight = new CANSparkFlex(INJEST_MOTOR_RIGHT, MotorType.kBrushless);
+
+		injestMotorBack = new CANSparkFlex(INJEST_MOTOR_BACK, MotorType.kBrushless);
+
+		resetMotors();
+	}
+
+	public void resetMotors() {
+		intakeMotorFront.restoreFactoryDefaults();
+		intakeMotorBack.restoreFactoryDefaults();
+		intakeMotorLeft.restoreFactoryDefaults();
+		intakeMotorRight.restoreFactoryDefaults();
+
+		injestMotorLeft.restoreFactoryDefaults();
+		injestMotorRight.restoreFactoryDefaults();
+
+		intakeMotorFront.setIdleMode(IdleMode.kBrake);
+		intakeMotorBack.setIdleMode(IdleMode.kBrake);
+		intakeMotorLeft.setIdleMode(IdleMode.kBrake);
+		intakeMotorRight.setIdleMode(IdleMode.kBrake);
+
+		injestMotorLeft.setIdleMode(IdleMode.kBrake);
+		injestMotorRight.setIdleMode(IdleMode.kBrake);
+
+		intakeMotorFront.setInverted(true);
+		intakeMotorBack.setInverted(true);
+		intakeMotorLeft.setInverted(true);
+		intakeMotorRight.setInverted(true);
+
+		injestMotorLeft.setInverted(true);
+		injestMotorRight.setInverted(true);
```

Comment:
To avoid this super-long function, it'd be better to create a private helper function (maybe `configureMotor(CANSparkFlex motor)`) that configures that one motor, and just have `resetMotors()` be a bunch of calls like this:
```
configureMotor(intakeMotorFront);
configureMotor(intakeMotorBack);
...
```

As long as the calls are in the same order as the class variable definitions above, it's easy to look at them to make sure all the motors are configured.  And in the future, the `configureMotor(..)` helper function can be updated to take parameters for any settings that end up being set to different values (such as current limit, inverted or not, etc.)

Reply:
All done, thanks for the suggestion